### ISSUE TITLE
AngleSharp broken on Xamarin.iOS?

### ISSUE THREAD

**Issue by corliss** [AngleSharp/AngleSharp#85]
NuGet gives errors upgrading a Xamarin.iOS project from 0.8.1 to 0.8.3;

```
System.InvalidOperationException: Could not install package 'AngleSharp 0.8.3'. You are trying to install this package into a project that targets 'Xamarin.iOS, Version=v1.0', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.
```

It looks like the new package does not support Xamarin.iOS. I noticed there are new dependencies on Microsoft.Bcl.Async - is that part of the reason?

Also, if 0.8.1 is entirely removed, then searching NuGet for AngleSharp in the context of the Xamarin.iOS project returns no results - further evidence that NuGet believes that AngleSharp does not support Xamarin.iOS.

Incidentally, I've recently switched to using Visual Studio 2015 RC. Not sure if that makes any difference.

**Comment by FlorianRappl** [comment]
I don't know. Unfortunately I have Xamarin not installed currently so I cannot investigate. But maybe someone else can do that. Only the .NET 4 and the SL variant have that dependency. The PCL does not have the dependency. I also don't know if the PCL is picked in case of Xamarin.iOS.

Any advices on how to fix / improve the NuGet package is highly welcome. I am shortly before releasing v0.8.4.

**Comment by corliss** [comment]
OK, I will investigate. But I'd like to know (since I don't have much experience with building NuGet packages):

a) Was there any work done specific to Xamarin.iOS in AngleSharp in the past, or did it work by accident?
b) Why is Microsoft.Bcl.Async needed by AngleSharp?

Edit: one more question
c) How do you currently build the NuGet package? Are there build instructions anywhere? I've built AngleSharp itself and run the unit tests but have not built the NuGet package so far.

**Comment by FlorianRappl** [comment]
a) There was no work done specific to Xamarin.iOS - it more or less works by accident since the Xamarin.iOS framework is compatible with the combination that forms the PCL used by AngleSharp (see [Xamarin Developer Guide](http://developer.xamarin.com/guides/cross-platform/application_fundamentals/pcl/introduction_to_portable_class_libraries/)).

b) The BCL async is required for `async` / `await`. The language part is only half of the story. Of course one needs a C# 5 (or higher) compiler, however, the language only transforms the code-block to use a state machine provided by the runtime. This state machine is not available in older versions of .NET - hence the requirement for the package. Again: This dependency is not required by the PCL, that "is" AngleSharp, but only for the legacy versions (.NET 4 and SL), which are also delivered by the same NuGet package.

c) Building the NuGet package is done via the `Package.ps1` PowerShell script. The last line (`$success = Publish-Package $version`) may be commented out, since publishing will currently only work with my personal NuGet account.

**Comment by corliss** [comment]
Cool. if you can go ahead and release v0.8.4 as you mentioned I will investigate using the latest code.

**Comment by Suchiman** [comment]
@corliss v0.8.4 has been released

**Comment by corliss** [comment]
I am now using a private build of AngleSharp because I need features currently not implemented that I've patched. As a result I am not using the NuGet package anymore. 

This bug still exists, however. If I get the chance I will try to investigate, but that may not happen anytime soon.

**Comment by FlorianRappl** [comment]
I now know that the bug is definitely related to NuGet and has been resolved with NuGet 3.0. In the long run AngleSharp will deploy a NuGet v3 package. That way a lot more platforms can be targeted, as dependencies are specified with fine-granularity.

I'll close this issue once the package has been deployed with NuGet v3 dependency declarations.
